packages/dds/tree/src/core/schema-stored/schema.ts

```diff
@@ -164,12 +184,14 @@ export abstract class TreeNodeStoredSchema {
 	protected _typeCheck!: MakeNominal;
 
 	/**
-	 * @privateRemarks
-	 * Returns TreeNodeSchemaDataFormat.
-	 * This is uses an opaque type to avoid leaking these types out of the package,
-	 * and is runtime validated by the codec.
+	 * Encode in the v1 schema format.
 	 */
-	public abstract encode(): TreeNodeSchemaDataFormat;
+	public abstract encodeV1(): TreeNodeSchemaDataFormatV1;
+
+	/**
+	 * Encode in the v2 schema format.
+	 */
+	public abstract encodeV2(): TreeNodeSchemaDataFormatV2;
```

Comment:
The return type of this method includes the metadata, but the implementations do not and rely on the caller to provide it.

I think it would be much more type safe to instead have this return V2's TreeNodeSchemaUnionFormat if you want the caller to add the metadata.